### ISSUE TITLE
The x0tigervncserver with offset data in option Geometry refuses to connect

### ISSUE THREAD

**Issue by Term84** [TigerVNC/tigervnc#1276]
**Describe the bug**
The x0tigervncserver with offset data in option Geometry refuses to connect.
If I try to set option -Geometry 200x200+0+0 the vnc client connects and works fine.
If I use x- or y- offsets e.g. "-Geometry 100x100+10+0" then the vnc client can't connect to the server.
In server's logs after start:

```
Thu Jun 17 12:35:11 2021
 Geometry:    Desktop geometry is set to 100x100+10+0
 XDesktop:    Using evdev codemap
 XDesktop:    
 XDesktop:    XTest extension present - version 2.2
 Main:        Listening on port 5905
3NI3X0 New X0tigervnc server 'vnc-srv:5 (admingrpc)' on port 5905 for display :0.
3NI3X0 Use xtigervncviewer -SecurityTypes None :5 to connect to the VNC server.
```

Server's logs after the connection attempt:

```
Thu Jun 17 12:35:23 2021
 Connections: accepted: [::1]::49740
 SConnection: Client needs protocol version 3.8
 SConnection: Client requests security type None(1)
 XDesktop:    Enabling 8 buttons of X pointer device
 XDesktop:    Allocated shared memory image
 VNCSConnST:  closing [::1]::49740: setPixelBuffer: invalid screen layout
 EncodeManager: Framebuffer updates: 0
 EncodeManager:   Total: 0 rects, 0 pixels
 EncodeManager:          0 B (1:-nan ratio)
 Connections: closed: [::1]::49740
```

**Client:**
 - OS: Linux Debian 11
 - VNC client: xtigervncviewer
 - VNC client version: 1.11.0
 - Client downloaded from: Linux distribution

**Server:**
 - OS: Linux Debian 11
 - VNC server: x0vncserver
 - VNC server version: 1.11.0
 - Server downloaded from: Linux distribution
 - Server was started using: x0tigervncserver :0 -rfbport 5905 -localhost yes -SecurityTypes None -Geometry 100x100+10+0

**Comment by CendioOssman** [comment]
I can reproduce that here. Let's see what might be wrong.

**Comment by CendioOssman** [comment]
It seems like the handling for this (added in 0fe375a209dd917de5cbabbfbe5915b9b58f8fa8) was always buggy. Should be fixed in 86a564a706d63a408817389ea509bf2be6de8315.

Thanks for reporting!
